src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php

```diff
@@ -220,15 +222,17 @@ public function startTest($test)
             if (isset($annotations['class']['expectedDeprecation'])) {
                 $test->getTestResultObject()->addError($test, new AssertionFailedError('`@expectedDeprecation` annotations are not allowed at the class level.'), 0);
             }
-            if (isset($annotations['method']['expectedDeprecation'])) {
-                if (!\in_array('legacy', $groups, true)) {
-                    $this->error = new AssertionFailedError('Only tests with the `@group legacy` annotation can have `@expectedDeprecation`.');
+            if (($hasExpectedDeprecationAnnotation = isset($annotations['method']['expectedDeprecation'])) || $this->checkNumAssertions = \in_array(SymfonyExpectDeprecationTrait::class, (new \ReflectionClass($test))->getTraitNames(), true)) {
+                if ($hasExpectedDeprecationAnnotation) {
+                    self::$expectedDeprecations = $annotations['method']['expectedDeprecation'];
+                    self::$previousErrorHandler = set_error_handler([$this, 'handleError']);
                 }
 
-                $test->getTestResultObject()->beStrictAboutTestsThatDoNotTestAnything(false);
+                if ($this->checkNumAssertions) {
```

Comment:
this won't be set if $hasExpectedDeprecationAnnotation is true, is that expected?

Reply:
The value will always be set since it has a default value.